lcm-logger/lcm_logger.c

```diff
@@ -276,7 +452,26 @@ static void *write_thread(void *user_data)
             logger->events_since_last_report = 0;
             logger->last_report_logsize = logger->logsize;
         }
-    }
+
+        // ---
+        // Disk Quota @Review: This whole section
+
+        // judfs: Should this be its own thread?
```

Comment:
One downside of the current approach:

If the user specifies a disk quota greater than what's available, they'll get a message that doesn't make a lot of sense:

```
$ build/lcm-logger/lcm-logger --disk-quota 1000G
There is currently 79.0 GiB free on disk.
lcm-logger will exit should that value reach 1000.0 GiB.
```

and it won't exit until the first message is logged. The behavior I would have expected is the logger would immediately exit (and possible also a different message).

A threaded approach might solve that, but an alternative would be to perform this check on startup.